src/helpers/activitypub/actor.unit.test.ts

```diff
@@ -226,3 +235,123 @@ describe('isHandle', () => {
+                .mockReturnValue(new URL('https://example.com/followers')),
+            sendActivity: vi.fn(),
+        } as unknown as RequestContext<ContextData>;
+    }
+
+    it('should return false if the site settings have not changed', async () => {
+        const db = {
+            get: vi.fn().mockResolvedValue({
+                id: 'https://example.com/user/1',
+                name: 'Site Title',
+                summary: 'Site Description',
+                preferredUsername: 'index',
+                icon: 'https://example.com/icon.png',
+                inbox: 'https://example.com/inbox',
+                outbox: 'https://example.com/outbox',
+                following: 'https://example.com/following',
+                followers: 'https://example.com/followers',
+                liked: 'https://example.com/liked',
+                url: 'https://example.com',
+            }),
+            set: vi.fn(),
+        } as unknown as KvStore;
+
+        const globaldb = {
+            get: vi.fn().mockResolvedValue(null),
+            set: vi.fn(),
+        } as unknown as KvStore;
+
+        const getSiteSettings = vi.fn().mockResolvedValue({
+            site: {
+                description: 'Site Description',
+                title: 'Site Title',
+                icon: 'https://example.com/icon.png',
+            },
+        });
+
+        const host = 'example.com';
+
+        const apCtx = mockApContext(db, globaldb);
+
+        const result = await updateSiteActor(apCtx, getSiteSettings, host);
+
+        assert(result === false);
+    });
+
+    it('Should update the site actor if the site settings have changed', async () => {
+        const db = {
+            get: vi.fn().mockResolvedValue({
+                id: 'https://example.com/user/1',
+                name: 'Site Title',
+                summary: 'Site Description',
+                preferredUsername: 'index',
+                icon: 'https://example.com/icon.png',
+                inbox: 'https://example.com/inbox',
+                outbox: 'https://example.com/outbox',
+                following: 'https://example.com/following',
+                followers: 'https://example.com/followers',
+                liked: 'https://example.com/liked',
+                url: 'https://example.com',
+            }),
+            set: vi.fn(),
+        } as unknown as KvStore;
+
+        const globaldb = {
+            get: vi.fn().mockResolvedValue(null),
+            set: vi.fn(),
+        } as unknown as KvStore;
+
+        const getSiteSettings = vi.fn().mockResolvedValue({
+            site: {
+                description: 'New Site Description',
+                title: 'New Site Title',
+                icon: 'https://example.com/icon.png',
+            },
+        });
+
+        const host = 'example.com';
+
+        const apCtx = mockApContext(db, globaldb);
+
+        const result = await updateSiteActor(apCtx, getSiteSettings, host);
+
+        assert(result === true);
```

Comment:
Should we make some assertions that the actual update happened with the values we expect + a message was published?